src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java

```diff
@@ -624,4 +627,31 @@ public void checkSkin(String adSkinLeftPath, String adSkinRightPath) {
     PageObjectLogging.log("SKIN", "SKIN presents on the page", true);
   }
 
+  /**
+   * Check if AdEngine loaded the ad web elements inside slot.
+   */
+  public boolean checkIfSlotOnPageLoaded(String slotName) {
+    changeImplicitWait(250, TimeUnit.MILLISECONDS);
+    try {
+      String slotSelector = AdsContent.getSlotSelector(slotName);
+      WebElement slot = driver.findElement(By.cssSelector(slotSelector));
+      if (slotName.equals(AdsContent.FLOATING_MEDREC)) {
+        triggerFloatingMedrec(slot);
+      }
+      List<WebElement> adWebElements = slot.findElements(By.cssSelector("div"));
+      return adWebElements.size() > 0;
+    } finally {
+      restoreDeaultImplicitWait();
+    }
+  }
+
+  private void triggerFloatingMedrec(WebElement slot) {
+    executeScript("window.scroll(0, 5000); setTimeout(function () {window.scroll(0, 5001) }, 500)");
+    // try to wait a floating medrec 3 seconds
+    try {
+      new WebDriverWait(driver, 3).until(CommonExpectedConditions.elementVisible(slot));
+    } catch (org.openqa.selenium.TimeoutException ignore) {
```

Comment:
you should log each situation with a separate message, or throw a meaningful exception - `FloatingMedrecNotAllowed`, cause if you want to trigger this element, and it's not available - it's an unexpected situation ASAIU